### PR TITLE
Jenkins: add iris optical flow SITL test

### PR DESCRIPTION
**Pull request by lamping7**
Initial step to get optical flow into CI.

Do we want to add this to ASan and coverage as well @dagar? The same question goes for MC_avoidance that was just merged in.

### CODE REVIEW

**Comment by dagar** [comment]
> Do we want to add this to ASan and coverage as well @dagar? The same question goes for MC_avoidance that was just merged in.

Yes, I was hoping to find a non-terrible way to keep them in sync (or a single source).

**Comment by lamping7** [comment]
There are a few issues to fix that are known so far:
- [ ] Jenkins doesn't have the sitl_gazebo lib for optical flow in the package
- [ ] The current mission test scripts require valid home and global position before starting
- [ ] Running the attitude control test and the position control test back to back crashes -- `[Err] [gazebo_mavlink_interface.cpp:405] bind failed: Address already in use, aborting`

If I work around all that, the position control test passes when I run this locally. However, the attitude test fails.
- [ ] x, y position is reset mid-flight for some unknown reason
Examples: 
- https://logs.px4.io/plot_app?log=2cc4baa7-5d93-4543-9304-0491aadc86fa (vehicle drops out of the sky in this one)
- https://logs.px4.io/plot_app?log=c3898a65-16b8-48dc-ad96-08b3335fc07e (thrust was increased from 70% to 90% to avoid falling)

I think the dropping altitude is unrelated. I haven't checked the mass associated with the sensor.

Where I see this:
```
header: 
  seq: 2398
  stamp: 
    secs: 82
    nsecs: 176464035
  frame_id: "map"
pose: 
  position: 
    x: 69.7704467773
    y: 34.4448814392
    z: 120.961540222
  orientation: 
    x: 0.120540329184
    y: -0.245032036052
    z: -0.0309482201273
    w: -0.961494495646
---
header: 
  seq: 2399
  stamp: 
    secs: 82
    nsecs: 208463064
  frame_id: "map"
pose: 
  position: 
    x: -1.48350703954e-14
    y: 2.98627098781e-30
    z: 121.137542725
  orientation: 
    x: 0.120590278468
    y: -0.245054747935
    z: -0.030939959345
    w: -0.961482694528
---
```

**Comment by lamping7** [comment]
Getting Jenkins to have access that that optical flow lib broke other stuff with the package. It seems like we'll have to have the lib in two different places. I'm not sure of all the players here. Number two above is fixed so far.

File: platforms/posix/CMakeLists.txt [hunk 1]
```diff
@@ -166,9 +166,18 @@ elseif ("${PX4_BOARD}" MATCHES "sitl")
 			${PROJECT_NAME}/build/px4_sitl_default
 		FILES_MATCHING
 			PATTERN "CMakeFiles" EXCLUDE
+			PATTERN "OpticalFlow" EXCLUDE
```
Comment (lamping7) [inline platforms/posix/CMakeLists.txt]: revert this

File: platforms/posix/CMakeLists.txt [hunk 2]
```diff
@@ -166,9 +166,18 @@ elseif ("${PX4_BOARD}" MATCHES "sitl")
...
 			PATTERN "*.so"
 		)
 
+	# sitl_gazebo built optical flow plugin
+	install(
+		FILES
+			${PROJECT_SOURCE_DIR}/build/px4_sitl_default/build_gazebo/OpticalFlow/libOpticalFlow.so
```
Comment (lamping7) [inline platforms/posix/CMakeLists.txt]: that makes it in the path for our test

**Comment by lamping7** [comment]
Number 1 is fixed.

**Comment by lamping7** [comment]
Number 3 involves waiting on the solution to https://github.com/PX4/sitl_gazebo/issues/289.

**Comment by lamping7** [comment]
Because of ports and gzserver not closing quick enough, the `make` commands that launch multiple tests back-to-back don't work anymore either (`tests_mission` and `tests_avoidance`). Ideas on a fix here @dagar @julianoes ?

For this test, I could split them out in Jenkins, but that doesn't solve the problem with this test file or the make targets.

**Comment by dagar** [comment]
> Because of ports and gzserver not closing quick enough, the `make` commands that launch multiple tests back-to-back don't work anymore either (`tests_mission` and `tests_offboard`). Ideas on a fix here @dagar @julianoes ?
> 
> For this test, I could split them out in Jenkins, but that doesn't solve the problem with this test file or the make targets.

Do you know if it's gzserver not closing quickly enough or questionable PX4 SITL shutdown/cleanup?

**Comment by julianoes** [comment]
That's probably related to https://github.com/PX4/sitl_gazebo/issues/289.

We could try to use `SO_REUSEPORT` as a workaround to allow binding to already bound port.

**Comment by julianoes** [comment]
This would still be nice!

**Comment by lamping7** [comment]
I just rebased. The Jenkins output looks different. I'll try this again.

**Comment by lamping7** [comment]
So it looks like my #3 above is no longer an issue, but we have that the position control is no longer passing. Attitude control still fails. Here are the logs from jenkins:

Attitude: https://logs.px4.io/plot_app?log=739b637d-945a-4e7e-a082-3a9984c21bd5
Position: https://logs.px4.io/plot_app?log=4029b2e6-bf33-48da-9a05-15f1599f0074

I got similar results when running this locally. Full console output can be found at: https://pastebin.com/jmqqg7U8

For the position test, I saw this locally, which doesn't show up in the logged messages in flight review from the Jenkins run:
```
WARN  [mc_pos_control] Auto activation failed with error: Activation Failed
WARN  [mc_pos_control] Position-Ctrl activation failed with error: Activation Failed
WARN  [mc_pos_control] Altitude-Ctrl activation failed with error: Activation Failed
WARN  [mc_pos_control] Position-Control Setpoint-Update failed
WARN  [mc_pos_control] Position-Control Setpoint-Update failed
```

**Comment by TSC21** [comment]
@lamping7 can we reboot this?

**Comment by lamping7** [comment]
Yes. I can rebase again and compare.

**Comment by lamping7** [comment]
I rebased just now. I'll check Jenkins in a bit and will edit this.

**Pull request closed without merging**